panos_upgrade_assurance/check_firewall.py

```diff
@@ -870,6 +864,185 @@ def check_ssl_cert_requirements(self, rsa: dict = {}, ecdsa: dict = {}) -> Check
         result.status = CheckStatus.SUCCESS
         return result
 
+    def check_non_finished_jobs(self) -> CheckResult:
+        """Check for any job with status different than FIN.
+
+        # Returns
+
+        CheckResult: Object of [`CheckResult`](/panos/docs/panos-upgrade-assurance/api/utils#class-checkresult) class taking \
+            value of:
+
+        * [`CheckStatus.SUCCESS`](/panos/docs/panos-upgrade-assurance/api/utils#class-checkstatus) when all jobs are in FIN state.
+        * [`CheckStatus.FAIL`](/panos/docs/panos-upgrade-assurance/api/utils#class-checkstatus) otherwise, `CheckResult.reason`
+            field contains information about the 1<sup>st</sup> job found with status different than FIN (job ID and the actual
+            status).
+
+        """
+        result = CheckResult()
+
+        all_jobs = self._node.get_jobs()
+
+        for jid, job in all_jobs.items():
+            if job["status"] != "FIN":
+                result.reason = f"At least one job (ID={jid}) is not in finished state (state={job['status']})."
+                return result
+
+        result.status = CheckStatus.SUCCESS
+        return result
+
+    def _calculate_time_distance(self, schedule_type: str, schedule: dict) -> (int, str):
```

Comment:
time difference might be a better wording instead of distance. Or we could say `_calculate_schedule_time_diff` and have an additional parameter as the reference date (instead of referring to `self._mp_now` within the function). This would also help the method be more generic for any other case.

```python
def _calculate_schedule_time_diff(self, reference_date: datetime, schedule_type: str, schedule: dict) -> (int, str):
```

Reply:
👍🏻